### PR TITLE
HeatMap chart bug fixes

### PR DESCRIPTION
**Pull request by Anush2303**
## New Behavior

Following issues are fixed for HeatMap chart:
1. hovercard coming as undefined.
2. chart coming greyed out.
3. string axis coming in alphabetical order.

image

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx [hunk 1]
```diff
@@ -714,7 +718,7 @@ export class HeatMapChartBase extends React.Component<IHeatMapChartProps, IHeatM
       if (this._yAxisType === YAxisType.DateAxis || this._yAxisType === YAxisType.NumericAxis) {
         return +a - +b;
       } else {
-        return a.toLowerCase() > b.toLowerCase() ? 1 : -1;
+        return 0;
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx:721]: don't change existing behavior. Add a new prop for sortOrder = Alphabetical, None. Use Alphabetical as default behavior.

Comment (Anush2303) [reply packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx:721]: okay, this might be the reason TCs are failing.

File: packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx [hunk 2]
```diff
@@ -247,7 +247,7 @@ export const DeclarativeChart: React.FunctionComponent<DeclarativeChartProps> =
       return (
         <HeatMapChart
           {...transformPlotlyJsonToHeatmapProps(plotlySchema)}
-          legendProps={legendProps}
+          legendProps={{ onChange: onActiveLegendsChange, canSelectMultipleLegends: false, selectedLegends: [] }}
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx:250]: heatmapchart doesn't need legendProps. It has hide legends = true

Comment (Anush2303) [reply packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx:250]: Previously, also we were passing legendProps. And now we have to pass empty array to selectedLegends prop.

File: change/@fluentui-react-charting-d2171258-9397-47d2-b063-9e1b96df4f2d.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-d2171258-9397-47d2-b063-9e1b96df4f2d.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv8 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33525&commitId=f85c863a5db1ff80a1c2be99f9a94c8d1c9471ba&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 4 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.Bottom Right Edge RTL.chromium.png|1593| Changed|
|Callout.Bottom left edge.chromium.png|2309| Changed|
|Callout.Beak 25.chromium.png|2186| Changed|
|Callout.Right bottom edge.chromium.png|3005| Changed|

</details>


  <details><summary><b>react-charting-AreaChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-AreaChart.Custom Accessibility.chromium.png|11| Changed|

</details>


  <details><summary><b>react-charting-HeatMapChart</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-HeatMapChart.Basic Dark Mode.chromium.png|5361| Changed|
|react-charting-HeatMapChart.Basic RTL.chromium.png|6720| Changed|
|react-charting-HeatMapChart.Basic.chromium.png|7095| Changed|

</details>


  <details><summary><b>react-charting-LineChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-LineChart.Gaps.chromium.png|1| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.types.ts [hunk 4]
```diff
@@ -113,6 +113,11 @@ export interface IHeatMapChartProps extends Pick<ICartesianChartProps, Exclude<k
    *@default false
    *Used for showing complete y axis lables   */
   showYAxisLables?: boolean;
+
+  /**
+   * The prop used to decide order of string axis labels
+   */
+  sortOrder?: string;
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.types.ts:120]: change to sortOrder?: 'none' | 'alphabetical'

File: packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx [hunk 5]
```diff
@@ -601,38 +601,48 @@ export class HeatMapChartBase extends React.Component<IHeatMapChartProps, IHeatM
...
+                datapoint.y = this._getStringFormattedNumber(datapoint.y as string, yAxisNumberFormatString);
+              }
+              if (this._yAxisType === YAxisType.StringAxis) {
+                datapoint.y = this._getFormattedLabelForYAxisDataPoint(datapoint.y as string);
+              }
+            })
+        : yPoints[item].forEach((datapoint: IHeatMapChartDataPoint) => {
+            if (this._xAxisType === XAxisTypes.StringAxis) {
+              datapoint.x = this._getFormattedLabelForXAxisDataPoint(datapoint.x as string);
+            }
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx:644]: this code looks redundant

Comment (Anush2303) [reply packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx:644]: No, this is not redundant. this._getFormattedLabelForXAxisDataPoint , this should run even if the sorting is not needed for 'None' case.

Comment (Anush2303) [reply packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx:644]: No, this is not redundant. this._getFormattedLabelForXAxisDataPoint, this should run even if sorting is not required in case of 'none' case.

File: packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.types.ts [hunk 6]
```diff
@@ -113,6 +113,11 @@ export interface IHeatMapChartProps extends Pick<ICartesianChartProps, Exclude<k
    *@default false
    *Used for showing complete y axis lables   */
   showYAxisLables?: boolean;
+
+  /**
+   * The prop used to decide order of string axis labels
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.types.ts:118]: >string [](http://example.com/codeflow?start=38&length=6)

add a comment about the default value

File: packages/charts/react-charting/src/components/HeatMapChart/__snapshots__/HeatMapChartRTL.test.tsx.snap [hunk 7]
```diff
@@ -1410,132 +1410,56 @@ exports[`HeatMapChart snapshot tests should render axis labels correctly When cu
           </g>
         </g>
         <g
-          aria-label="xPoint p3, yPoint p2. Nasty, 25."
+          aria-label="xPoint p3, yPoint p2. , No data available."
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HeatMapChart/__snapshots__/HeatMapChartRTL.test.tsx.snap:1413]: >data [](http://example.com/codeflow?start=49&length=4)

why this change

Comment (Anush2303) [reply packages/charts/react-charting/src/components/HeatMapChart/__snapshots__/HeatMapChartRTL.test.tsx.snap:1413]: need to see, I just updated snapshots so that build passes.

File: packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx [hunk 8]
```diff
@@ -601,38 +601,45 @@ export class HeatMapChartBase extends React.Component<IHeatMapChartProps, IHeatM
...
-        });
+      this.props.sortOrder !== 'none'
+        ? yPoints[item]
+            .sort((a: IHeatMapChartDataPoint, b: IHeatMapChartDataPoint) => {
+              if (this._xAxisType === XAxisTypes.StringAxis) {
+                return this.props.sortOrder === 'none'
+                  ? 0
+                  : (a.x as string).toLowerCase() > (b.x as string).toLowerCase()
+                  ? 1
+                  : -1;
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx:612]: why do you need any change other than these in this function?

File: packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx (hunk 2 above)
Comment (AtishayMsft) [reply packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx:250]: there are no legends in heatmap in declarative chart. Why do you need legend props then?

File: packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx [hunk 9]
```diff
@@ -244,14 +244,7 @@ export const DeclarativeChart: React.FunctionComponent<DeclarativeChartProps> =
...
     case 'heatmap':
-      return (
-        <HeatMapChart
-          {...transformPlotlyJsonToHeatmapProps(plotlySchema)}
-          legendProps={legendProps}
-          componentRef={chartRef}
-          calloutProps={{ layerProps: { eventBubblingEnabled: true } }}
-        />
-      );
+      return <HeatMapChart {...transformPlotlyJsonToHeatmapProps(plotlySchema)} />;
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx:247]: why remove componentRef and calloutProps

**Review by AtishayMsft — APPROVED** [review]
